util/builtin_operators.go

```diff
@@ -240,19 +240,32 @@ func GlobMatchFunc(args ...interface{}) (interface{}, error) {
 
 // GenerateGFunction is the factory method of the g(_, _) function.
 func GenerateGFunction(rm rbac.RoleManager) govaluate.ExpressionFunction {
+	memoized := map[string]bool{}
```

Comment:
I think we need to set a default capacity for this `cache` to avoid DOS attack

Reply:
when will this cache be collected by gc?

after enforce function call?

Reply:
> I think we need to set a default capacity for this `cache` to avoid DOS attack

What is DOS attack?

Reply:
> when will this cache be collected by gc?
> 
> after enforce function call?

I add cleanup in g.

Reply:
@nodece I'm afraid of having infinite growing cache. DOS can easily fill up all the memory.

However, seems your closure should be collected after enforce call so maybe it's fine, but still we need to limit the capacity

Reply:
@nodece fix typo: memoized -> memorized

Reply:
@GopherJ  I can't confirm the capacity.

Reply:
@nodece finally I think this `cleanup` is not necessary because closure will be collected after `enforce` function call.

How do you think? @UnderTreeTech @hsluoyz

Reply:
I'm not sure if collected.